### DISCUSSION TITLE
About spanish translation...

### DISCUSSION THREAD

**Discussion by thiagol-fenix** [diasurgical/DevilutionX#5994] [General]
Hello everyone!.

First of all, thanks for the translation work realized by the community. I hope we can discuss here in a cordial and respectful manner how to improve and polish the Spanish translation of this wonderful game/project.

I am Spaniard and a big fan of the first two installments of the saga. I would like to contextualize my experience with this translation.

I've been playing for the last week and it's not hard to see that there's a lot of work to be done in general (particularly in the nomenclature and description of objects).

So I've been running some tests of my own and I want to share with you some very brief conclusions:

- The translation mixes Latin American Spanish with Spanish from Spain. This generates inconsistencies and I think it can be a problem. Maybe choosing only one of the two or separating the translation into two... I would go for the second option.
- The object naming system is completely broken. The prefixes of the original English version are not possible in Spanish, they would have to be transformed into first or second suffixes (like Diablo II approach). I have made progress on this point on my own.
- The base items (armors, shields, weapons, etc) should be consistent with the sequel because the translation work on Diablo II was very professional and I think it would be a perfect fit here.

Having said all this, I would like to collaborate in the translation. The third point I was talking about before (basic items) I have it ready (Diablo II 1.14d approach) and, if you allow it, I would like to share it with you.

That's all, I hope that together we can make an amazing translation!.

Cheers! :D

**Comment by AJenbo** [comment]
Hello and welcome,

>     * The translation mixes Latin American Spanish with Spanish from Spain. This generates inconsistencies and I think it can be a problem. Maybe choosing only one of the two or separating the translation into two... I would go for the second option.

Sounds good, I would suggest making the current one es-ES. But of cause that depends on the dialect spoken by whom ever is available to do the works.

>     * The object naming system is completely broken. The prefixes of the original English version are not possible in Spanish, they would have to be transformed into first or second suffixes (like Diablo II approach). I have made progress on this point on my own.

You can do this by changing this translation line:

```po
#. TRANSLATORS: Constructs item names. Format: {Prefix} {Item} of {Suffix}. Example: King's Long Sword of the Whale
#: Source/items.cpp:1161
msgid "{0} {1} of {2}"
msgstr "{0} {1} de {2}"
```
to this
```po
#. TRANSLATORS: Constructs item names. Format: {Prefix} {Item} of {Suffix}. Example: King's Long Sword of the Whale
#: Source/items.cpp:1161
msgid "{0} {1} of {2}"
msgstr "{1} {0} {2}"
```
(there are a few other similar lines that you also might want to adjust)

> I would like to share it with you.

Please provide changes and updates via Pull Requests.

If you haven't already then please see the https://github.com/diasurgical/devilutionX/wiki/Translation-Guide

Others who have worked on the Spanish translation in the past are @hiperiondev, @b3t0787, and @Mad-Soft. Maybe they have some further input.

Lastly we are currently gearing up to do a new release and there are currently 21 untranslated strings for the Spanish translation so if you could start with translating these that would be appreciated.

**Reply by thiagol-fenix** [reply]
Thanks, as soon as I can I will try to make a pull request for the part of basic objects I mentioned before.


> You can do this by changing this translation line:
> 
> ```po
> #. TRANSLATORS: Constructs item names. Format: {Prefix} {Item} of {Suffix}. Example: King's Long Sword of the Whale
> #: Source/items.cpp:1161
> msgid "{0} {1} of {2}"
> msgstr "{0} {1} de {2}"
> ```
> 
> to this
> 
> ```po
> #. TRANSLATORS: Constructs item names. Format: {Prefix} {Item} of {Suffix}. Example: King's Long Sword of the Whale
> #: Source/items.cpp:1161
> msgid "{0} {1} of {2}"
> msgstr "{1} {0} {2}"
> ```
> 
> (there are a few other similar lines that you also might want to adjust)

That's exactly what I've been doing these days, making adjustments and testing it in the game and the result is pretty good!... but there are complications: 


- Combination of preposition "of" + determiner "the" ("... of the ...")

This combination in Spanish takes into account gender and plurals, going from one to four possible combinations: "del", "de la", "de los", "de las". A few examples:

Short Sword of the eagle  <-> Espada Corta del aguila
Gothic Shield of the moon <-> Escudo Gótico de la luna
Long Staff of the heavens <-> Bastón Largo de los cielos
Breast Plate of the starts   <-> Coraza Blindada de las estrellas

My temporary solution for this has been to add by hand the preposition-determiner combo corresponding to each suffix... very laborious and inelegant, but it works.

- The gender of the noun (object name) determines the gender of the adjective (prefix).

This problem is much more complicated and I can't think of a solution... probably requires touching code. A few examples:

Jagged Long Sword    <-> Espada Larga Dentada
Jagged Hunter's Bow  <-> Arco de Cazador Dentado
White Small Shield      <-> Escudo Pequeño Blanco
White Robe                 <-> Túnica Blanca


Anyway, first I will try to make the pull request.

Cheers! :D

**Reply by AJenbo** [reply]
@glebm it's issues like these that makes me think that adding support for grammatical gender isn't enough to solve translation issues and why all of that might be better handled by a scripting system where each language can define there own name generation function.

**Reply by AJenbo** [reply]
> My temporary solution for this has been to add by hand the preposition-determiner combo corresponding to each suffix... very laborious and inelegant, but it works.

Ok, but keep in mind that we are thinking about adding a more dynamic way of handling this, which would require things to be reworked again.

> This problem is much more complicated and I can't think of a solution... probably requires touching code.

Yeah, I think we will handle this via support for scripting translation for each language individually. Preferably the engine should be fairly language agnostic.

**Comment by hiperiondev** [comment]
Hi to all

The problem of the Latin languages was dealt with at the beginning of the Spanish translation and, as @Ajembo says, the solution would be some scripting system.
There is also the issue of the use of ellipses such as the Saxon genitive that does not exist in the Latin languages.
Regarding the "style" of Spanish it is more complicated, many countries, many styles, many idioms... Spanish is a complicated language :-)
It could be separated by countries, if someone is willing to correct all the texts!
Unifying the names of items and monsters with Diablo II would be a good idea. There should be a glossary first to be able to verify the equality in the translations.
